test/cbmc/proofs/TCP/prvTCPPrepareSend/TCPPrepareSend_harness.c

```diff
@@ -62,12 +62,24 @@ NetworkBufferDescriptor_t * pxGetNetworkBufferWithDescriptor( size_t xRequestedS
     return pxBuffer;
 }
 
+/* Get rid of configASSERT in FreeRTOS_TCP_IP.c */
+BaseType_t xIsCallingFromIPTask( void ) 
+{
+    return pdTRUE;
+}
+
 void harness()
 {
-    FreeRTOS_Socket_t * pxSocket = ensure_FreeRTOS_Socket_t_is_allocated();
+    FreeRTOS_Socket_t * pxSocket = malloc( sizeof( FreeRTOS_Socket_t ) );
+    __CPROVER_assume( pxSocket != NULL );
+    pxSocket->u.xTCP.rxStream = malloc( sizeof( StreamBuffer_t ) );
+    pxSocket->u.xTCP.txStream = malloc( sizeof( StreamBuffer_t ) );
+    pxSocket->u.xTCP.pxPeerSocket = malloc( sizeof( FreeRTOS_Socket_t ) );
+
     NetworkBufferDescriptor_t * pxNetworkBuffer = ensure_FreeRTOS_NetworkBuffer_is_allocated();
     size_t socketSize = sizeof( FreeRTOS_Socket_t );
-    size_t bufferSize = sizeof( TCPPacket_t );
+    /* Allocates min. buffer size required for the proof */
```

Comment:
This comment just tells me what is written in the code right below.

Ideally, comments should answer questions like why does the proof need exactly `sizeof( TCPPacket_t ) + uxIPHeaderSizeSocket( pxSocket )` and not just `sizeof( TCPPacket_t )`.

We could improve on these comments.